plotly/dashboard_objs/dashboard_objs.py

```diff
@@ -0,0 +1,302 @@
+                    path_to_box.append(first_or_second)
+                    for key in self.box_ids_dict.keys():
+                        if self.box_ids_dict[key] == path_to_box:
+                            self.box_ids_dict.pop(key)
+
+                    try:
+                        max_id = max(self.box_ids_dict.keys())
+                    except ValueError:
+                        max_id = 0
+                    self.box_ids_dict[max_id + 1] = path_to_box
+
+    def _get_box(self, box_id):
+        """Returns box from box_id number."""
+
+        loc_in_dashboard = self['layout']
+        for path in self.box_ids_dict[box_id]:
+            loc_in_dashboard = loc_in_dashboard[path]
+        return loc_in_dashboard
+
+    def get_preview(self):
+        """
+        Returns JSON and HTML respresentation of the dashboard.
+
+        HTML coming soon to a theater near you.
+        """
+        # print JSON figure
+        pprint.pprint(self)
+
+    def insert(self, box, box_id=None, side='above'):
+        """
+        The user-friendly method for inserting boxes into the Dashboard.
+
+        box: the box you are inserting into the dashboard.
+        box_id: pre-existing box you use as a reference point.
+        """
+        # doesn't need box_id or side specified
+        if 'first' not in self['layout']:
+            self._insert(Container(), [])
+            self._insert(box, ['first'])
+        else:
+            if box_id is None:
+                raise exceptions.PlotlyError(
+                    "Make sure the box_id is specfied if there is at least "
+                    "one box in your dashboard."
+                )
+            if box_id not in self.box_ids_dict:
+                raise exceptions.PlotlyError(
+                    "Your box_id must a number which is pointing to a box in "
+                    "your dashboard."
+                )
+
+            if side == 'above':
+                old_box = self._get_box(box_id)
+                self._insert(
+                    Container(box, old_box, direction='vertical'),
+                    self.box_ids_dict[box_id]
+                )
+            elif side == 'below':
+                old_box = self._get_box(box_id)
+                self._insert(
+                    Container(old_box, box, direction='vertical'),
+                    self.box_ids_dict[box_id]
+                )
+            elif side == 'left':
+                old_box = self._get_box(box_id)
+                self._insert(
+                    Container(box, old_box, direction='horizontal'),
+                    self.box_ids_dict[box_id]
+                )
+            elif side == 'right':
+                old_box = self._get_box(box_id)
+                self._insert(
+                    Container(old_box, box, direction='horizontal'),
+                    self.box_ids_dict[box_id]
+                )
+
+
+def upload_dashboard(dashboard_object, filename, world_readable,
```

Comment:
This is just a semantic thing right, since a dashboard is JSON and a `Dashboard` object is also JSON (a `dict`)? The renaming is for clarity in what can be passed, correct?